### PR TITLE
Fail if ManagedFields is empty on apply

### PR DESCRIPTION
**Pull request by joshmsamuels**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/cli-runtime/pkg/resource/helper.go [hunk 1]
```diff
@@ -142,6 +144,21 @@ func (m *Helper) Patch(namespace, name string, pt types.PatchType, data []byte,
 	if options == nil {
 		options = &metav1.PatchOptions{}
 	}
+
+	obj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, data)
```
Comment (apelisse) [inline staging/src/k8s.io/cli-runtime/pkg/resource/helper.go]: That patch can be in many formats, I doubt you can decode them with this function.

File: staging/src/k8s.io/cli-runtime/pkg/resource/helper.go [hunk 2]
```diff
@@ -17,10 +17,12 @@ limitations under the License.
 package resource
 
 import (
+	"errors"
```
Comment (apelisse) [inline staging/src/k8s.io/cli-runtime/pkg/resource/helper.go]: This is changing `cli-runtime`, which only protects client-side for clients that use `cli-runtime`. If we don't want this to happen,  shouldn't it be done on the server?

**Review by apelisse — COMMENTED** [review]
I'm not sure I understand the rational? Why do we want to prevent patches to managedFields?

File: staging/src/k8s.io/cli-runtime/pkg/resource/helper.go (hunk 1 above)
Comment (joshmsamuels) [reply staging/src/k8s.io/cli-runtime/pkg/resource/helper.go]: So how would you decode the patch? Is there a reference that has all the acceptable formats of `data`?

File: staging/src/k8s.io/cli-runtime/pkg/resource/helper.go (hunk 2 above)
Comment (joshmsamuels) [reply staging/src/k8s.io/cli-runtime/pkg/resource/helper.go]: yup I had changes in the client and server and accidentally pushed the client ones

File: staging/src/k8s.io/cli-runtime/pkg/resource/helper.go (hunk 1 above)
Comment (apelisse) [reply staging/src/k8s.io/cli-runtime/pkg/resource/helper.go]: I don't think you really can

**Pull request closed without merging**